src/store/selectors/subjectsPicker.js

```diff
@@ -0,0 +1,51 @@
+const subjectsPicker = ({ primary, lingual }) => {
+  const maxScoreWrapper = {}
+  const primaryArray = primary.filter(subject => subject.active)
+  const lingualArray = lingual.filter(subject => subject.active)
+
+  let maxPrimarySubject = _findBestSubject(primaryArray)
+  let maxLingualSubject = _findBestSubject(lingualArray)
+
+  // This one is for humane formula only, if the best lingual score is better than primary score, we remove it
+  // from lingual subjects and treat as primary, then look for new (second) best lingual.
+  // If the lingual * 4 + second best lingual is better than primary * 4 + lingual, we got the highest possible score,
+  // else we return wrapper with regular primary and lingual as they provide better score
+
+  if (maxLingualSubject.maxScore.value * 4 > maxPrimarySubject.maxScore.value) {
+    const filteredLingualArray = lingual.filter(subject => subject !== maxLingualSubject)
+
+    let maxPrimarySubject2 = maxLingualSubject
+    let maxLingualSubject2 = _findBestSubject(filteredLingualArray)
+
+    if (maxPrimarySubject2.maxScore.value * 4 + maxLingualSubject2.maxScore.value >
+      maxPrimarySubject.maxScore.value * 4 + maxLingualSubject.maxScore.value) {
+      maxScoreWrapper.primary = maxPrimarySubject2
+      maxScoreWrapper.lingual = maxLingualSubject2
+
+      return maxScoreWrapper
+    }
+  }
+
+  maxScoreWrapper.primary = maxPrimarySubject
+  maxScoreWrapper.lingual = maxLingualSubject
+
+  return maxScoreWrapper
+}
+
+const _findBestSubject = (subjectsArray) => {
+  let bestSubject = {
+    name: '',
+    maxScore: {
+      value: 0,
+      formula: ''
+    }
+  }
+
+  subjectsArray.forEach((subject) => {
```

Comment:
This is rather for exercise sake than required change.

Think about how you could use `subjectsArray.reduce` method to get the best subject. (`_findBestSubject ` implementation takes 3 lines and do not require defining/reassigning `bestSubject`)

BTW `.reduce` is awesome and it's good to get the solid understanding of it.

Reply:
I've seen your implementation with Array.prototype.reduce and it's really great, but the fact I use bestSubject 'empty' object is for API reasons. I've got for example:
```
const maxPrimaryScore = maxScoreWrapper.primary.maxScore.value
```
Maybe some `if(maxScoreWrapper.primary.maxScore && maxScoreWrapper.primary.maxScore.value)` would suffice, but I didn't want to change the external logic when modyfing the subjectsPicker and getCourseScore functions.

Reply:
Ok, but `bestSubject` can be used as initial data for the reduce.

Reply:
So I believe this is the solution: 
```
const _findBestSubject = (subjectsArray) => {
  subjectsArray.reduce((bestSubject, subject) => {
    if (subject.maxScore.value > bestSubject.maxScore.value) return subject
  }, {
    name: '',
    maxScore: {
      value: 0,
      formula: ''
    }
  })

  return subjectsArray[0]
}
```

Reply:
Have you tried if it works? ;]

Reply:
I see my mistake. I thought if the callback function returns undefined, the previousValue stays untouched (and is the last previousValue). My test subjectsArray was with values like [100, 200] and this was totally not sufficient. 😄  

So I think this one is ok:
```js
_findBestSubject = (subjectsArray) => {
  return subjectsArray.reduce(
    (bestSubject, subject) => {
      if (subject.maxScore.value > bestSubject.maxScore.value) return subject
      else return bestSubject
    },
    {
      name: '',
      maxScore: {
        value: 0,
        formula: ''
      }
    }
  )
}
```